### DISCUSSION TITLE
In-line editing of fields on index page

### DISCUSSION THREAD

**Discussion by jimarick** [laravel/nova-issues#4229] [Ideas & Feature Requests]
I would love to for nova to ship with an option to edit text and Boolean fields on the index page of a resource. Perhaps users could apply the ->inlineEdit() method to a field, or even an $inlineEdit property to the resource class.

Then on the resource index page there could be an inline edit toggle to turn it on and off. This would allow users to change text fields, or toggle Booleans on and off from the index. 

I'm aware of some third party packages providing similar functionality, but something first party would be really nice. 

I have a database of thousands of questions, and it would be really nice to quickly scroll through them all clicking them as true/false on the index page, rather than going into the edit page of each item.

**Comment by javier-itg** [comment]
Yes. This is much needed. It could also be depending on the user/policy. _If admin, then enable inline edit._ 

Specially in development and testing, you need to activate and deactivate configurations constantly. A toggle directly on the index is the best approach; 1 click, done! Currently, you have to:

1. Scroll/go to the edit icon and click.
2. Wait for the new page to load. 
3. Scroll to the field you wanted to change. 
4. Change the field.
5. Scroll down to the update button and click.
6. Wait for the page to reload. 
7. Find your way back to the index.

Too much.

**Comment by andristet** [comment]
+1
